extensions/precompiles/actions.go

```diff
@@ -96,15 +100,16 @@ func (p *ProcedureContext) Values() map[string]any {
 
 // NewScope creates a new procedure context for a child procedure.
 // It will not inherit the values or last result from the parent.
-// It will inherit the dbid and procedure from the parent.
+// It will inherit the dbid, procedure, and stack depth from the parent.
 func (p *ProcedureContext) NewScope() *ProcedureContext {
 	return &ProcedureContext{
-		Ctx:       p.Ctx,
-		Signer:    p.Signer,
-		Caller:    p.Caller,
-		values:    make(map[string]any),
-		DBID:      p.DBID,
-		Procedure: p.Procedure,
+		Ctx:        p.Ctx,
+		Signer:     p.Signer,
+		Caller:     p.Caller,
+		values:     make(map[string]any),
+		DBID:       p.DBID,
+		Procedure:  p.Procedure,
+		StackDepth: p.StackDepth,
```

Comment:
Initially I had this with a `+1` from the parent procedure's context, but `(*baseDataset).Call` also does `NewScope` so it can get incremented twice unintentionally.